landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+        grain size distribution can be assigned to the initial soil layer and
+        the bedrock layer
+        """
+
+        if not is_bedrock_distribution_flag:
+
+            is_bedrock_distribution_flag = self._is_bedrock_distribution_flag
+            median_size = self._initial_median_size
+            std = self._std
+            total_soil_weight = self._initial_total_soil_weight
+            grains_weight__distribution = self._generate_normal_distribution(
+                median_size=median_size, std=std, total_soil_weight=total_soil_weight
+            )
+
+        else:
+            total_bedrock_weight = 10000  # There is no meaning for this number
+            if median_size is None:
+                grains_weight__distribution = self.g_state0
+            else:
+                median_size = median_size
+                if std is None:
+                    std = self._CV * median_size
+                else:
+                    std = std
+                grains_weight__distribution = self._generate_normal_distribution(
+                    median_size=median_size,
+                    std=std,
+                    total_soil_weight=total_bedrock_weight,
+                )
+
+        if not is_bedrock_distribution_flag:
+            self.g_state = (
+                np.ones(
+                    (
+                        int(np.shape(self._grid)[0]),
+                        int(np.shape(self._grid)[1]),
+                        int(len(grains_weight__distribution)),
+                    )
+                )
+                * grains_weight__distribution
+            )
+
+            self.g_state0 = grains_weight__distribution
+            self._grid.at_node["grains__weight"][self._grid.core_nodes, :] = 1
+            self._grid.at_node["grains__weight"] *= grains_weight__distribution
+            layer_depth = np.sum(self.g_state0) / (
+                self._soil_density * self._grid.dx * self._grid.dx
+            )
+            layer_depth /= 1 - self._phi
+
+            self._grid.at_node["soil__depth"][self._grid.core_nodes] += layer_depth
+            self._grid.at_node["topographic__elevation"] = (
+                self._grid.at_node["soil__depth"]
+                + self._grid.at_node["bedrock__elevation"]
+            )
+
+            self.g_state_bedrock = grains_weight__distribution
+            self._grid.at_node["bed_grains__proportions"][self._grid.core_nodes] = 1
+            self._grid.at_node["bed_grains__proportions"][
+                self._grid.core_nodes
+            ] *= np.divide(self.g_state_bedrock, np.sum(self.g_state_bedrock))
+
+        else:
+            self.g_state_bedrock = grains_weight__distribution
+            self._grid.at_node["bed_grains__proportions"][self._grid.core_nodes] = 1
+            self._grid.at_node["bed_grains__proportions"][
+                self._grid.core_nodes
+            ] *= np.divide(self.g_state_bedrock, np.sum(self.g_state_bedrock))
```

Comment:
This looks like repeated code. How about removing one of these blocks and then dedenting the other?